content/3_Silver/Prefix_Sums_2.mdx

```diff
@@ -102,11 +102,98 @@ export const problems = {
 
-<Spoiler title="Why are the two methods equivalent?">
+<Spoiler title="Solution">
+
+### Solution
+
+Suppose that we store the prefix sums at each index in the array. Intuitively, the maximum subarray sum ending at each index is the difference between the maximum prefix sum and the minimum prefix sum. Thus, we can iterate each element in the array and store the minimum prefix sum up to the current index. Then, we can find the maximum subarray sum by subtracting the current prefix sum to the minimum prefix sum.
+
+C++ code:
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+using ll = long long;
+using vl = vector<ll>;
+
+int main(){
+	int n; cin >> n;
+	vl dp(n+1, 0); // Prefix sum array initially filled with 0's
+	for(int i=1; i<=n; i++){
+		ll x; cin >> x;
+		dp[i] = dp[i-1] + x; // Compute the prefix sum at each element
+	}
+	ll max_subarray_sum = dp[0], min_prefix_sum = dp[0];
+	for(int i=1; i<=n; i++){
+		// The maximum subarray sum is the maximum difference the current prefix sum and the minimum prefix sum up to the current index
+		max_subarray_sum = max(max_subarray_sum, dp[i] - min_prefix_sum);
+		min_prefix_sum = min(min_prefix_sum, dp[i]);
+	}
+	cout << max_subarray_sum << endl;
+}
+```
+
+</CPPSection>
+
+</LanguageSection>
+
+</Spoiler>
+
+<Warning>
+
+Note: It is recommended that you skip the following solution.
+
+</Warning>
+
+<Spoiler title = "Alternative Solution">
+
+### Solution
+
+Suppose we fix the first element of a subarray. The Kadane's Algorithm solution involves starting a new subarray sum at an element if that element is greater than the current subarray sum with an addition to the current element. If the subarray sum ending at the current element is greater than the current element, then we continue the subarray sum. Otherwise, we set the new subarray sum equal to the current element and restart our subarray sum. To find the maximum subarray sum over all possible sums, we store an integer $S$, representing the maximum subarray sum at each iteration of the array. When we finish iterating the array, we print $S$.
+
+Watch this [video](https://www.youtube.com/watch?v=jnoVtCKECmQ&t=611s) for a more detailed explanation of the approach above.
+
+C++ code:
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+using ll = long long;
+using vl = vector<ll>;
+
+int main(){
+	int n; cin >> n;
+	vl x(n);
+	for(int i=0; i<n; i++){
+		cin >> x[i];
+	}
+	ll current_sum = x[0], max_subarray_sum = x[0];
+	for(int i=1; i<n; i++){
+		current_sum = max(current_sum + x[i], x[i]); 
+	// Continue the subarray sum or start a new subarray sum beginning at the current element
```

Comment:
shorten or put this on two lines (also fix the indentation)

Reply:
Yep

Reply:
Should I remove the tab or should I add another tab??

Reply:
Add another tab